manager/orchestrator/restart/restart.go

```diff
@@ -214,8 +219,8 @@ func (r *Supervisor) shouldRestart(ctx context.Context, t *api.Task, service *ap
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	restartInfo := r.history[instanceTuple]
-	if restartInfo == nil {
+	restartInfo := r.historyByService[t.ServiceID][instanceTuple]
```

Comment:
No, we don't have a concept of an instance ID. We track instances as a combination of service ID, and either a slot number or node ID.